src/internal_rep.rs

```diff
@@ -386,6 +392,23 @@ impl TypeInfo {
         }
         false
     }
+
+    pub fn get_all_parent_names<'a>(&'a self, types: &'a TypeMap) -> Vec<&'a CascadeString> {
+        let mut ret = Vec::new();
```

Comment:
Why not use a `BTreeSet` for `ret`? `derived_body` already use this type and it would make it easier to check for duplicate values.